### PR TITLE
MVTec 3D and Folder3D

### PR DESCRIPTION
**Pull request by alexriedel1**
Hi,
it's the fix of the failing tests of https://github.com/openvinotoolkit/anomalib/pull/907.

The dataloader always expects mask paths even if no mask paths are provided. I removed the dummy insertion of the mask paths and now added it back again.

I think its debatable whether the dataloader should always load the mask paths, but thats another PR I guess:
https://github.com/openvinotoolkit/anomalib/blob/78ab50e1cb1e613180da9f9a4bf371367e6a5ce5/anomalib/data/base/dataset.py#L105-L118

### CODE REVIEW

File: anomalib/data/folder.py [hunk 1]
```diff
@@ -137,31 +69,45 @@ def make_folder_dataset(
...
+        assert (
+            samples.loc[samples.label_index == 1]
+            .apply(lambda x: Path(x.image_path).stem in Path(x.mask_path).stem, axis=1)
+            .all()
+        ), "Mismatch between anomalous images and mask images. Make sure the mask files \
+            folder follow the same naming convention as the anomalous images in the dataset \
+            (e.g. image: '000.png', mask: '000.png')."
+    else:
+        samples["mask_path"] = ""
+
```
Comment (alexriedel1) [inline anomalib/data/folder.py:106]: Thats the actual new part

**Review by djdameln — APPROVED** [review]
Thanks for the fix. I agree that we should not load the paths when the task type is classification. Let's merge this for now and address it in a separate PR.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks for the efforts! This is a great addition

**Comment by alexriedel1** [comment]
Sorry for the trouble, I wasn't able to run all the tests on my windows machine..

The datamodule now passes all the test in `tests\pre_merge\datasets\test_synthetic_data.py` and `tests\pre_merge\datasets\test_datamodule.py`:
`tests\pre_merge\datasets\test_synthetic_data.py .....       [100%]`
`tests\pre_merge\datasets\test_datamodule.py ...........     [100%]`


that previously failed:
```
FAILED tests/pre_merge/datasets/test_datamodule.py::TestSubsetSplitting::test_abnormal_dir_omitted_from_dir
FAILED tests/pre_merge/datasets/test_datamodule.py::TestSubsetSplitting::test_abnormal_dir_omitted_synthetic
FAILED tests/pre_merge/datasets/test_synthetic_data.py::test_create_synthetic_dataset
FAILED tests/pre_merge/datasets/test_synthetic_data.py::test_create_from_dataset
```

I really wanted to remove the iteration over the dataframe in the folder dataset:
https://github.com/openvinotoolkit/anomalib/blob/566e8cb4a3e36cad383f69c2be696bad512e2c59/anomalib/data/folder.py#L153-L158

This is even warned about in the pandas docs https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iterrows.html
```
You should never modify something you are iterating over. This is not guaranteed to work in all cases. Depending on the data types, the iterator returns a copy and not a view, and writing to it will have no effect.
```

**Comment by samet-akcay** [comment]
Thanks @alexriedel1! Much appreciated!

**Pull request merged**